### PR TITLE
Adding config option to set xhr useXDR param

### PR DESCRIPTION
**Pull request by benmanu**

### CODE REVIEW

**Comment by markstuart** [comment]
What is the requirement for setting this in config? My understanding is that the xhr library will only use XDR on IE9 if set to true. Why would you need to explicitly set to false?

**Comment by markstuart** [comment]
I like the idea of being able to override the request information though (headers etc), so it might work to do something like this:

```js
  "dataSource": {
    "request": {
      "url": "..." // override other options here: https://github.com/Raynos/xhr
    }
    "type": "longPoll",
    "refresh": 10000
  }
```

```js
var merge = require('merge')

module.exports = (config, cb) => {

  var xhrOptions = { useXDR: true }
  xhrOptions = merge(xhrOptions, config.request)

  xhr(xhrOptions, function (err, resp, body) {
  //...
```

**Comment by benmanu** [comment]
We ran into some issues in IE9 where with this value set to 'true' our data source wasn't being returned, where setting the value to 'false' gave correct results (referencing a local .json file). I like your approach with setting the default xhrOptions and overriding through config, I'll incorporate that solution into the branch.

File: README.md [hunk 1]
```diff
@@ -202,7 +202,9 @@ Manually tested in:
         "dataSource": {
             "url": "...",
             "type": "longPoll",
-            "refresh": 10000
+            "refresh": 10000,
+            "useXDR": true || false // see https://github.com/Raynos/xhr#optionsusexdr for usage
+                                    // if undefined will default to true
```
Comment (markstuart) [inline README.md]: I think the request related options (url and useXDR in this example) need to move into a object of their own if we are using the 'overriding default options' pattern for `xhrOptions`. In this flattened structure the `xhrOptions` would end up including the `type` and `refresh` properties as well once the config object was merged onto the default object (https://github.com/mediasuitenz/mappy/pull/46/files#diff-c385ce3d4304f2e93613b211f96a38b9R14), which aren't valid options to `xhr`.

Namespacing the request related config into a separate object would be much safer I think, similar to my example from yesterday.

**Comment by digitalsadhu** [comment]
@markstuart wheres this one at?

**Comment by markstuart** [comment]
Was going to test and merge today :)

**Comment by digitalsadhu** [comment]
awesome

**Comment by markstuart** [comment]
Merged via commandline at a1b2643

**Pull request closed without merging**